Initial public draft — Sigmoid Rebase Function Proposal (#14)
ampleforth/AIPs: AIPs/aip-5.md

Below we'll quickly review the basic sigmoid, and then explain the rationale for the "balanced" sigmoid proposed.

#### 1. Sigmoid
A sigmoid function takes the following shape, notice the horizontal asymptotes and the maximum slope at the target price($1):

<p align="center">
<img src="../assets/aip-5/basic_sigmoid_latex_thin.jpg" width="700"/>
</p>

**1.1. _Sigmoid Function and Parameters_**

This equation takes, `x`, the normalized difference between `VWAP` and the price target. It returns, `Y`, the corresponding supply change percentage.

```
Y = supply change %
x = normalized price deviation
```

<p align="center">
<img src="../assets/aip-5/basic_sigmoid_eq_2.png" alt="drawing" width="380"/>
</p>

> It has shaping parameters that determine: lower asymptote, upper asymptote, and the steepness of the curve (ie: growth rate) around its origin.

Author: good catch, merging this in manually

```
L = lower asymptote
U = upper asymptote
B = growth rate
```

#### 2. "Balanced" Sigmoid

> Although the Sigmoid is a good start, we can improve upon it by scaling supply changes such that they “balance” one another. More specifically:

Reviewer: The modification applied here is not a scale modifier but the shape for the contraction side is constructed differently. 
```suggestion
Although the Sigmoid is a good start, we can improve upon it by modifying the supply changes such that they “balance” one another. More specifically:
```

Author: can we say transforming?

* If a demand-change-factor of `A` corresponds with supply-change-factor `B`.
* We want to enforce that a demand-change-factor of `1/A` corresponds with a supply-change-factor of `1/B`. 

> This way, supply reactions to equal and opposite relative changes in demand, always even eachother out. For more context, see the alternating series example in the motivation section above.

Reviewer: typo:
```suggestion
This way, supply reactions to equal and opposite relative changes in demand, always even each other out. For more context, see the alternating series example in the motivation section above.
```

Author: merging in

</p>

*_<sup>\* In the equation above, P denotes normalized price, P = vwap_price/target</sup>_*

#### 3. "Balanced" Sigmoid vs Linear

We expect that the “balanced” sigmoid supply curve will cause the Ampleforth network to react symmetrically to relative demand signals, and eliminate the bias towards prolonged contraction periods.

<p align="center">
<img src="../assets/aip-5/combined_latexchart_withdeviation.png" alt="drawing" width="700"/>
</p>

### Technical Specification
<!--The technical specification should outline the public API of the changes proposed. That is, changes to any of the interfaces Ampleforth currently exposes or the creations of new ones.-->

This update creates no changes to external APIs. Clients, including exchanges, who listen to AMPL’s rebase events still receive the absolute supply change integer as before. However, note that any external application that calculates the delta independently will need to update their calculation logic.

#### Initial Parameter Values

These parameters were set to achieve the outcomes stated above (see motivation)—while keeping the aggregate expansion rate as similar to the current protocol as possible.

> Recall that the contraction curve is derived from the expansion curve through the "balancing" solution above. Thus, we need only verify that the shaping parameters selected, produce an expansion curve that generally resembles the current rebasing function. Thereafter, a balanced contraction curve will follow from the balanced_sigmoid equation as defined above. 

Reviewer: "Thereafter" implies time dependency which doesn't exist
```suggestion
Recall that the contraction curve is derived from the expansion curve through the "balancing" solution above. Thus, we need only verify that the shaping parameters selected, produce an expansion curve that generally resembles the current rebasing function. based on it a balanced contraction curve is constructed from the balanced_sigmoid equation as defined above. 
```
